querysrv.go

```diff
@@ -329,10 +329,12 @@ func (s *querysrv) handleAnnounce(ctx context.Context, remote net.IP, deviceID p
 		}
 
 		ip := net.ParseIP(host)
-		if len(ip) == 0 || ip.IsUnspecified() {
-			uri.Host = net.JoinHostPort(remote.String(), port)
+		if host == "" || ip.IsUnspecified() {
```

Comment:
Judging by [go lang source](https://golang.org/src/net/ip.go), `ip.IsUnspecified` returns `false` for nil `ip`